src/Plugin/Action/SystemPageRedirect.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\rules\Plugin\Action\SystemPageRedirect.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\Core\Routing\RedirectDestinationInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Provides "Page redirect" rules action.
+ *
+ * @Action(
+ *   id = "rules_page_redirect",
+ *   label = @Translation("Page redirect"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "url" = @ContextDefinition("uri",
+ *       label = @Translation("URL"),
+ *       description = @Translation("A Drupal path, path alias, or external URL to redirect to. Enter (optional) queries after ? and (optional) anchor after #."),
+ *     ),
+ *     "force" = @ContextDefinition("boolean",
+ *       label = @Translation("Force redirect"),
+ *       description = @Translation("Force the redirect even if another destination parameter is present. Per default Drupal would redirect to the path given as destination parameter, in case it is set. Usually the destination parameter is set by appending it to the URL."),
+ *     ),
+ *     "destination" = @ContextDefinition("boolean",
+ *       label = @Translation("Append destination parameter"),
+ *       description = @Translation("Whether to append a destination parameter to the URL, so another redirect issued later on would lead back to the origin page."),
+ *     ),
+ *   },
+ *   provides = {
```

Comment:
So as far as I can see we will have to create a Response listener that will issue the redirect if it was ordered by the action execution.